### PR TITLE
feat: support for dictionaries

### PR DESCRIPTION
**Pull request by joelabrahamsson**
### What kind of change does this PR introduce? (check at least one)
 - [ ] Bugfix
 - [x ] Feature
 - [ ] Test
 - [ ] Docs
 - [ ] Refactor
 - [ ] Build-related changes
 - [ ] Other, please describe:

### Description:

This PR adds support for dictionaries with objects as values as described in the OpenAPI spec under "Value type" (second example here: https://swagger.io/docs/specification/data-models/dictionaries/
As my understanding of the codebase is limited I don't think the implementation is optimal but hopefully someone with more familiarity of the codebase can refactor it to a satisfactory level?

### CODE REVIEW

File: transforms/components/index.js [hunk 1]
```diff
@@ -64,6 +64,17 @@ const getRequiredProperties = properties => (
 
 const formatRequiredProperties = requiredProperties => requiredProperties.map(getPropertyName);
 
+const addDictionaryAdditionalProperties = typedef => {
+  if (typedef.type?.applications?.length === 2 && typedef.type.applications[0].name === 'string') {
```
Comment (LonelyPrincess) [inline transforms/components/index.js:68]: Seems like this line is causing tests to break. 😢 

```bash
SyntaxError: Unexpected token '.'
```

Apparently the reason is that [the optional chaining (`?.`) operator is not supported on Node versions inferior to 14](https://stackoverflow.com/questions/59574047/how-to-use-optional-chaining-in-node-js-12), and the tests are currently expected to run in both Node 10 and 12.

**Review by LonelyPrincess — CHANGES_REQUESTED** [review]
Hey, this is a very interesting feature! 👏🏻 

I've requested a few changes to make the syntax clearer and fix the tests (which are failing at the moment), but once this is addressed, I think this would be a nice addition to the project. 👍🏻

File: test/transforms/components/index.test.js [hunk 2]
```diff
@@ -847,4 +847,48 @@ describe('parseComponents method', () => {
...
+        * Profile
+        * @typedef {object} Profile
+        *
+        * @property {string} email
+        */
+      `,
+    `
+       /**
+        * Profiles dict
+        * @typedef {Object.<string, Profile>} Profiles
```
[the remark this answers is not in the document]
Comment (joelabrahamsson) [reply test/transforms/components/index.test.js:863]: Seems like a good idea to me, however `Object.<string, Profile>` is compliant with standard JSDoc while I guess `Dictionary<Profile>` wouldn't be. I'm not sure if that matters from this project's perspective though?

[the remark this answers is not in the document]
Comment (LonelyPrincess) [reply test/transforms/components/index.test.js:863]: For what I'm aware, this library doesn't strictly follow the JSDoc standard, so I suppose it wouldn't be much of a problem. 🤔 

Let me show you a very clear example of that: the syntax for required and optional fields. This is how they're defined in this library:

```
@property {string} title.required - Required parameter
@property {string} artist - Optional field
```

In the standard JSDoc syntax, however, optional and required fields are represented differently:

```
@property {string} title - Required parameter
@property {string} [artist] - Optional field
```

[the remark this answers is not in the document]
Comment (joelabrahamsson) [reply test/transforms/components/index.test.js]: Thanks for the guidance! I have updated the PR.

**Review by LonelyPrincess — APPROVED** [review]

**Comment by kevinccbsg** [comment]
@LonelyPrincess thanks a lot for all your help reviewing the code! and thanks @joelabrahamsson for the contribution

@all-contributors please add @joelabrahamsson for code

**Pull request merged**